R/zchunk_LA154.Transport.R

```diff
@@ -32,71 +31,215 @@ module_gcam.usa_LA154.Transport_DISABLED <- function(command, ...) {
+      # We will create a list first, concatenating EIA-fuel and -sector, so as to selectively remove those pairs from the dataset
+      trnUCD_EIA_mapping %>%
+        mutate(fuel_sector = paste(EIA_fuel, EIA_sector)) %>%
+        .[["fuel_sector"]] %>%
+        unique() ->
+        list_fuel_sector
+
+      # Here is the state-level data for which to calculate state shares
+      # We will first filter for only relevant EIA-fuel and -sector pairs
+      L101.EIA_use_all_Bbtu %>%
+        filter(year %in% HISTORICAL_YEARS) %>% # Ensure within historical period
+        mutate(fuel_sector = paste(EIA_fuel, EIA_sector)) %>% # Create concatenate list in base dataframe to match the syntax of our list above
+        filter(fuel_sector %in% list_fuel_sector) %>% # Filtering for just EIA-fuel/sector pairs
+        rename(value_state = value) %>%
+        select(state, EIA_fuel, EIA_sector, sector, fuel, year, value_state) ->
+        EIA_transportation_state
+
+      # To calculate the state share, we need to calculate the national amount
+      EIA_transportation_state %>%
+        group_by(EIA_fuel, EIA_sector, sector, fuel, year) %>% # Dropping state
+        summarise(value_national = sum(value_state)) %>%
+        ungroup() ->
+        EIA_transportation_national
+
+      # Now the state shares can be calculated by dividing the state data by the national
+      EIA_transportation_state %>%
+        left_join_error_no_match(EIA_transportation_national, by = c("EIA_fuel", "EIA_sector", "sector", "fuel", "year")) %>%
+        mutate(value_share = value_state / value_national) %>% # Calculating state's share
+        # NAs were introduced where national values were 0. Replace NAs with zeros.
+        replace_na(list(value_share = 0)) %>%
+        select(state, EIA_fuel, EIA_sector, year, value_share) ->
+        EIA_transportation_state_share
+
+      # The full USA tran UCD database can now be apportioned to the states
+      # A list of states is created for when expanding the transportation table to include states
+      list_states <- unique(L100.Pop_thous_state$state)
+
+      # Creating the first of the three output tables
+      Transportation_energy_consumption %>%
+        repeat_add_columns(tibble::tibble(state = list_states)) %>%
+        left_join_error_no_match(EIA_transportation_state_share, by = c("state", "EIA_fuel", "EIA_sector", "year")) %>%
+        mutate(value = value * value_share) %>% # Allocating across the states
+        select(state, UCD_sector, mode, size.class, UCD_technology, UCD_fuel, fuel, year, value) ->
+        L154.in_EJ_state_trn_m_sz_tech_F
+
+      # As a final step, aggregate by fuel and name the sector
+      # This creates the second of three output tables
+      L154.in_EJ_state_trn_m_sz_tech_F %>%
+        group_by(state, fuel, year) %>%
+        summarise(value = sum(value)) %>%
+        ungroup() %>%
+        mutate("sector" = "transportation") %>%
```

Comment:
This is actually creating a column named "sector" and filling it with "transportation" entries. I took the quotes off "sector" and made a comment.